checkov/sast/checks_infra/checks_parser.py

```diff
@@ -149,3 +165,34 @@ def _get_definitions_list_items(self, definitions: List[Dict[str, Any]]) -> List
                 conf.append(self._parse_definition(definition))
 
         return conf
+
+    def _parse_taint_mode_definition(self, definitions: Dict[str, Any]) -> Dict[str, Any]:
+        conf: Dict[str, Any] = {
+            str(SemgrepAttribute.PATTERN_SOURCES): [],
+            str(SemgrepAttribute.PATTERN_SINKS): []
+        }
+        if not isinstance(definitions, list):
+            raise TypeError(f'bad taint mode definition type, got {type(definitions)} instead of list')
+        for definition in definitions:
+            cond_type = definition.get('cond_type', '')
+            if not cond_type:
+                raise AttributeError('BQL policy is missing a condition type')
+            operator = definition.get('operator', '')
+            if not operator:
+                raise AttributeError(f'BQL policy condition type: {cond_type} is missing an operator')
+            definition_value = definition.get('value')
+            if not definition_value:
+                raise AttributeError(f'BQL policy condition type: {cond_type} is missing a definition value')
+
+            if cond_type == BqlConditionType.PATTERN_SOURCE:
```

Comment:
switch case ?

Reply:
no switch case in python ❤️

Reply:
```From version 3.10 upwards, Python has implemented a switch case feature called “structural pattern matching”. You can implement this feature with the match and case keywords.```

from 3.10 :( 
let's start sunsetting old python versions :)